main.go

```diff
@@ -144,19 +147,22 @@ func main() {
 	restConfig.QPS = float32(restConfigQPS)
 	restConfig.Burst = restConfigBurst
 	mgr, err := ctrl.NewManager(restConfig, ctrl.Options{
-		Scheme:                  scheme,
-		MetricsBindAddress:      metricsBindAddr,
-		Port:                    webhookPort,
+		Scheme:             scheme,
+		MetricsBindAddress: metricsBindAddr,
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Port: webhookPort,
+		}),
 		LeaderElection:          enableLeaderElection,
 		LeaderElectionID:        "baremetal-operator",
 		LeaderElectionNamespace: leaderElectionNamespace,
-		Namespace:               watchNamespace,
 		HealthProbeBindAddress:  healthAddr,
-
-		NewCache: cache.BuilderWithOptions(cache.Options{
-			SelectorsByObject: secretutils.AddSecretSelector(nil),
-		}),
-	})
+		Cache: cache.Options{
+			Namespaces: []string{watchNamespace},
+		},
+		NewCache: func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
```

Comment:
According to https://github.com/kubernetes-sigs/controller-runtime/commit/54e84fcff3c29412e77799426eb35f7c8aaf679f the `BuilderWithOptions()` function was removed because we don't have to provide `NewCache` any more. We can just set:

```
   ByObject: secretUtils.AddSecretSelector(nil),
```

in the cache.Options just above.